server/src/main/java/org/eclipse/lsp/cobol/core/visitor/LanguageSplittingListener.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2021 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+
+package org.eclipse.lsp.cobol.core.visitor;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.antlr.v4.runtime.*;
+import org.antlr.v4.runtime.misc.Interval;
+import org.antlr.v4.runtime.tree.ParseTreeListener;
+import org.eclipse.lsp.cobol.core.CobolParserBaseListener;
+import org.eclipse.lsp.cobol.core.Db2SqlLexer;
+import org.eclipse.lsp.cobol.core.Db2SqlParser;
+import org.eclipse.lsp.cobol.core.model.EmbeddedCode;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+import static org.eclipse.lsp.cobol.core.CobolParser.*;
+
+/**
+ * This lexer separates embedded language code parts from the main document and parse them with
+ * specific parsers.
+ */
+@AllArgsConstructor
+public class LanguageSplittingListener extends CobolParserBaseListener {
```

Comment:
Optional: Do you consider to rename `EmbeddedLanguagesListener`.   Using embeddedCodeParts internally.

Reply:
I agree to rename. I didn't understand what do you mean in "Using embeddedCodeParts internally."

Reply:
> I agree to rename. I didn't understand what do you mean in "Using embeddedCodeParts internally."

I mean field: embeddedCodePart